op-alt-da/damgr.go

```diff
@@ -117,8 +117,16 @@ func (d *DA) OnFinalizedHeadSignal(f HeadSignalFn) {
 func (d *DA) updateFinalizedHead(l1Finalized eth.L1BlockRef) {
 	d.l1FinalizedHead = l1Finalized
 	// Prune the state to the finalized head
-	d.state.Prune(l1Finalized.ID())
-	d.finalizedHead = d.state.lastPrunedCommitment
+	lastPrunedCommIncBlock := d.state.Prune(l1Finalized.ID())
+	d.log.Debug("updateFinalizedHead", "currFinalizedHead", d.finalizedHead.Number, "lastPrunedCommIncBlock", lastPrunedCommIncBlock.Number, "l1Finalized", l1Finalized.Number)
+	// If a commitment was pruned, set the finalized head to that commitment's inclusion block
+	// When no commitments are left to be pruned (one example is if we have failed over to ethda)
+	// then updateFinalizedFromL1 becomes the main driver of the finalized head.
+	// Note that updateFinalizedFromL1 is only called when len(d.state.commitments) == 0
+	var zero eth.L1BlockRef
+	if lastPrunedCommIncBlock != zero {
```

Comment:
nit: use zero value directly, to be consistent with other zero value cases in the repo

Reply:
[34b2df9](https://github.com/ethereum-optimism/optimism/pull/12845/commits/34b2df99e2a44f55727c13c7eef6d5f683d2916a)